### PR TITLE
Mission board

### PR DESCRIPTION
**Pull request by datadanne**
Closes RIG-59, RIG-60, RIG-61, RIG-62, RIG-64

### CODE REVIEW

File: garage/src/abis/MissionsManager.ts
Comment (sanderpick) [inline garage/src/abis/MissionsManager.ts:1]: would it be cleaner to have this as part of the rigs package vs. manual abi?

File: ethereum/contracts/MissionsManager.sol
Comment (sanderpick) [inline ethereum/contracts/MissionsManager.sol:72]: i remember chatting about this... so the front end does some validation / construction of `data`, but ultimately `data` could be anything cause this method is open to the world. we'll just manually look at the entries and throw out any junk, right?

**Review by sanderpick — DISMISSED** [review]
Whoa, tons of work here! LGTM. @dtbuchholz can you give this a skim as well? lots of additions.

File: ethereum/contracts/MissionsManager.sol
Comment (datadanne) [reply ethereum/contracts/MissionsManager.sol:72]: Yeah exactly. Contributions need to be manually reviewed by an admin and won't show in the Garage UI until they've been approved, so you can't spam a bunch of junk and make it look ugly

File: garage/src/abis/MissionsManager.ts
Comment (datadanne) [reply garage/src/abis/MissionsManager.ts:1]: Yes definitely. We actually export the generated types from typechain in the rigs package but we can't use them, I don't remember if it is because they are just incompatible with wagmi or if it has something to do with the way they are exported from the rigs package but that would definitely be much nicer

File: ethereum/test/MissionsManager.ts
Comment (dtbuchholz) [inline ethereum/test/MissionsManager.ts:124]: 👀 [1337](https://1337skulls.xyz/)

File: garage/src/hooks/usePersistentState.ts
Comment (dtbuchholz) [inline garage/src/hooks/usePersistentState.ts:25]: should these values be "test"? or does it not really matter what they are?

File: garage/src/components/SignManifestoModal.tsx
Comment (dtbuchholz) [inline garage/src/components/SignManifestoModal.tsx:119]: when a user agrees to the terms, i don't think that data is stored in a table, correct? it'd be a nice little FT bonus we could track in the `missions` table. if you sign the pledge, you get a little FT bump—the `max_number_of_contributions` could be capped at 3000 (1:1 with the # of Rigs).

File: garage/src/pages/Admin/MissionAdmin.tsx
Comment (dtbuchholz) [inline garage/src/pages/Admin/MissionAdmin.tsx:184]: should the admin contribution approval process work in this PR? i tried both accepting/rejecting with the [admin wallet](https://discord.com/channels/592843512312102924/1119061737073811596/1148514993290084374) in the [admin UI](https://rigs-garage-6t8sxo896-tableland.vercel.app/admin/missions/1), and neither worked. some screenshots:
- attempting to accept (note: both the reject/accept buttons spin...not a big deal but calling it out in case it helps troubleshoot):
Screenshot 2023-09-11 at 3 07 41 PM

- error shown after waiting for the approve tx to finalize: 
Screenshot 2023-09-11 at 3 08 21 PM


- and upon refreshing, the second row (the one i tried to approve) is still pending: Screenshot 2023-09-11 at 3 16 53 PM

File: ethereum/contracts/MissionsManager.sol
Comment (dtbuchholz) [reply ethereum/contracts/MissionsManager.sol:72]: @datadanne i verified the contract on [mumbai](https://mumbai.polygonscan.com/address/0xb169b0bed3e4ca014dc329318935e0dcf50e14b6) to test out the "anyone can write" aspect (no issues with the logic!). lmk if you want me to push the commit; it just updates the `verify.ts` script.

File: garage/src/hooks/usePersistentState.ts
Comment (datadanne) [reply garage/src/hooks/usePersistentState.ts:25]: yeah it doesn't matter what the values are, it is just testing if localStorage is available and falling back to a cookie if it isn't. also ht gpt for writing this

File: garage/src/components/SignManifestoModal.tsx
Comment (datadanne) [reply garage/src/components/SignManifestoModal.tsx:119]: Nope, we just store it in localStorage/cookies. I agree that it would be nice to store it in tableland but I don't think it makes sense unless we can pay the gas for the txn: feels weird to limit users from being able to see the missions & contribute unless they pay for it (most have already paid to buy/mint a rig already), and some might not have eth on arbitrum so they would have to bridge first as well

File: garage/src/pages/Admin/MissionAdmin.tsx
Comment (datadanne) [reply garage/src/pages/Admin/MissionAdmin.tsx:184]: hm yes that should work, I'll take a look

Comment (datadanne) [reply garage/src/pages/Admin/MissionAdmin.tsx:184]: ah, I forgot to give the admin wallet permission to update the contributions table. hm, maybe I should update the contract & have approvals go through the contract instead, that way we don't have to remember to give it both the admin role and those permissions

Comment (datadanne) [reply garage/src/pages/Admin/MissionAdmin.tsx:184]: https://mumbai.polygonscan.com/tx/0x71078318effcf5a6946db104e6fc9be84f2628d56856a1e957e0afd7b9293202

File: ethereum/contracts/MissionsManager.sol
Comment (datadanne) [reply ethereum/contracts/MissionsManager.sol:72]: > @datadanne i verified the contract on [mumbai](https://mumbai.polygonscan.com/address/0xb169b0bed3e4ca014dc329318935e0dcf50e14b6) to test out the "anyone can write" aspect (no issues with the logic!). lmk if you want me to push the commit; it just updates the `verify.ts` script.

nice! yes, please push that commit :pray:

File: garage/src/components/SignManifestoModal.tsx
Comment (dtbuchholz) [reply garage/src/components/SignManifestoModal.tsx:119]: Fair point. It'd be cool if we supported gasless txs for this, but that obviously adds unneeded complexity.

File: garage/src/pages/Admin/MissionAdmin.tsx
Comment (dtbuchholz) [reply garage/src/pages/Admin/MissionAdmin.tsx:184]: Nice!

File: garage/src/abis/MissionsManager.ts
Comment (dtbuchholz) [reply garage/src/abis/MissionsManager.ts:1]: ✅

File: ethereum/contracts/MissionsManager.sol
Comment (dtbuchholz) [reply ethereum/contracts/MissionsManager.sol:72]: ✅

**Comment by datadanne** [comment]
@dtbuchholz @sanderpick can I get a re-approval on this PR? the reviews got dismissed when dtb pushed a commit. I also added three commits: 
- a tiny bug fix, I noticed today that the 'contributions pending reivew' count was off because of a missing `WHERE mission_id = ?` clause in the query ([4c992c2](https://github.com/tablelandnetwork/rigs/pull/919/commits/4c992c2a3be86ab288e4eeeabdf6a9aa3226bb7e))
- I reverted the temp commit that hardcoded the staging deployment for testing purposes ([7504bf8](https://github.com/tablelandnetwork/rigs/pull/919/commits/7504bf8a529cc73952d68c30c6d82b8ccb20b0a4))
- I made a small improvement to the contribution review form so that it doesn't crash the page if the data is incorrectly formatted, thx @dtbuchholz for doing some proper testing :raised_hands:  ([d0e0aec](https://github.com/tablelandnetwork/rigs/pull/919/commits/d0e0aec6c2c151e189b6d42a177ad9c9f8375c2a))

--

The deployment won't work because this PR needs the `mission`-related tables/contracts to be part of the rigs package. Lmk how you want to go about deploying this, should I still merge it to staging or do you want to publish a pre-release of the package first @sanderpick?

**Review by dtbuchholz — DISMISSED** [review]

**Comment by sanderpick** [comment]
let's merge it to staging and I can work on the deployment of the package / contract / tables

**Comment by datadanne** [comment]
> let's merge it to staging and I can work on the deployment of the package / contract / tables

Ah actually I can't haha, it requires all checks to pass. I can add back the temp change that hardcodes the staging dev into `env.ts` and update the SHIP linear issue with a check to remember to revert that? or can you override the status checks?

image

**Comment by sanderpick** [comment]
ah yeah. i can just do the package release now and then you can make the changes here. one sec

**Comment by sanderpick** [comment]
pushed 0.3.3-pre0

**Review by sanderpick — APPROVED** [review]

**Pull request merged**
